src/driving/vehicle_boat.py

```diff
@@ -0,0 +1,276 @@
+
+import math
+
+import arcade
+
+import constants
+from driving.drive_mode import DriveMode
+
+DEFAULT_WORLD_SIZE_X = constants.SCREEN_WIDTH
+DEFAULT_WORLD_SIZE_Y = constants.SCREEN_HEIGHT
+DEFAULT_ACCELERATION_RATE = 2.5
+DEFAULT_DECELERATION_RATE = 0.1
+DEFAULT_BRAKE_RATE = 25
+DEFAULT_FRICTION = 25
+DEFAULT_DRIVE_SPEED = 12
+DEFAULT_TURN_SPEED = 2.2
+
+DEFAULT_CONTROL_LOCKOUT = 0.4
+
+IMPACT_FORCE_FLOOR = 1.5
+
+AXIS_FLOOR = 0.3
+
+
+class Boat(DriveMode):
+    def __init__(self, controls: MovementControls):
+        self.name = "Boat"
+        self.move_controls = controls
+
+        self.vehicle_velocity = (0, 0, 0)
+        self.external_velocity = (0, 0, 0)
+
+        self.friction = DEFAULT_FRICTION
+
+        self.drive_speed = DEFAULT_DRIVE_SPEED
+        self.turn_speed = DEFAULT_TURN_SPEED
+        self.debug_world_boundary_x = DEFAULT_WORLD_SIZE_X
+        self.debug_world_boundary_y = DEFAULT_WORLD_SIZE_Y
+        self.impact_buffer = 0
+
+        self.in_collision = False
+
+        self.sail_height = 0  # 0:furled 1:unfurled
+        self.raise_sail_speed = DEFAULT_DECELERATION_RATE
+        self.wheel_position = 0  #:-1 full left 1:full right
+        self.turn_wheel_speed = DEFAULT_DECELERATION_RATE  #:-1 full left 1:full right
+        self.raise_anchor_speed = 0.5
+        self.drop_anchor_speed = 2
+        self.anchor_position = 0  #  0:up 1:down
+
+        self.wind_assist = 1
+
+    def set_stats(self, drive, turn, acc, brake):
+        self.acceleration_rate = acc
+        self.brake_rate = brake
+
+        self.drive_speed = drive
+        self.turn_speed = turn
+
+    def drive_input(self, delta_time: float, vehicle: Vehicle, input: PlayerInput):
+
+        x = 0
+        y = 0
+        turn = 0
+        sail_change = 0
+        wheel_change = -input.x_axis.value * self.turn_wheel_speed
+        anchor_change = 0
+
+        if input.accelerate_axis.value > 0.1:
+            sail_change = input.accelerate_axis.value * DEFAULT_ACCELERATION_RATE
+
+        if input.brake_axis.value > 0.1:
+            sail_change = input.brake_axis.value * -self.raise_sail_speed
+
+        if input.ry_axis.value > 0.1:
+            anchor_change = input.ry_axis.value * -self.raise_anchor_speed * delta_time
+        else:
+            # if the anchor isnt raised all the way it should fall back to the 'down' position
+            if input.ry_axis.value < -0.1 or self.anchor_position > 0:
+                anchor_change = delta_time * self.drop_anchor_speed
+
+        self.anchor_position = self.clamp(self.anchor_position + anchor_change, 0, 1)
+
+        self.wheel_position = self.clamp(
+            self.wheel_position + (wheel_change * delta_time), -1, 1
+        )
+        self.sail_height = self.clamp(
+            self.sail_height + (sail_change * delta_time), 0, 1
+        )
+
+        vehicle_angle = vehicle.location[2]  # + (turn * delta_time)
+        self.set_wind_assist(math.degrees(vehicle.location[2]) % 360)
```

Comment:
I bet the math for [tacking into the wind](https://en.wikipedia.org/wiki/Tacking_(sailing)) would be "fun".